src/check-nsfw.ts

```diff
@@ -1,48 +1,45 @@
-const got = require('got');
-const tensorflow = require('@tensorflow/tfjs-node');
-const nsfw = require('nsfwjs');
-const decodeGif = require('decode-gif');
-const jpeg = require('jpeg-js');
-const Discord = require('discord.js');
-const db = require('./db');
-const config = require('../config.json');
-
-let model;
+import got from 'got';
+import * as tensorflow from '@tensorflow/tfjs-node';
+import { load } from 'nsfwjs';
+import decodeGif from 'decode-gif';
+import jpeg from 'jpeg-js';
+import { ChannelType, EmbedBuilder, TextChannel } from 'discord.js';
+import { getDB } from '@/db';
+import path from 'path';
+import type { Tensor3D } from '@tensorflow/tfjs-node';
+import type { AttachmentPayload, Message } from 'discord.js';
+import type { NSFWJS, predictionType } from 'nsfwjs';
+import config from '@/config.json';
+
+let model: NSFWJS;
 const GIF_MAGIC = Buffer.from([0x47, 0x49, 0x46, 0x38]);
 
-/**
- * 
- * @param {Discord.Message} message 
- * @param {Array} urls
- */
-async function checkNSFW(message, urls) {
-	if (message.channel.nsfw) {
+export async function checkNSFW(message: Message, urls: string[]): Promise<void> {
+	if (message.channel instanceof TextChannel && message.channel.nsfw) {
 		return; // Do not check if the channel is NSFW
 	}
 
 	if (!model) {
-		let modelPath;
-
-		if (config.quantized_nsfw_model) {
-			modelPath = 'file://' + __dirname + '/nsfw_model/quantized/';
-		} else {
-			modelPath = 'file://' + __dirname + '/nsfw_model/normal/';
-		}
+		const quantized = config.quantized_nsfw_model ? 'quantized' : 'normal';
+		const modelPath = path.join(path.resolve(config.nsfw_model_path), quantized);
+		const modelUri = `file://${modelPath}/`;
 		
-		model = await nsfw.load(modelPath, { type: 'graph' });
+		// * We have to set the options to `any` here because although `type` is a valid option
+		// * it's not included in the types, this is also an inline type so we can't enhance it
```

Comment:
Are you looking at the right version? We're using 2.4 at the moment and [this](https://github.com/infinitered/nsfwjs/blob/v2.4.0/src/index.ts#L33) is what the load function is like there. Basically, it uses the default to set a pretty strict inline type which ends up looking like this in the dist:
```ts
export declare function load(base?: string, options?: {
    size: number;
}): Promise<NSFWJS>;
```
Looking at this we could probably override the whole type of that function, but maybe it's better to just upgrade the library and get rid of all of the hackiness? That's probably better done as part of #16 though

Reply:
I was looking at the latest version in the `master` branch, which seems to be significantly different than what you have there <https://github.com/infinitered/nsfwjs/blob/4a81f6a09dc9e25cdb95d675849f3678ec59274e/src/index.ts#L115-L125>

Probably because we are ***2 major versions*** behind. The latest is 4.1.0. I was not aware we were so far behind

This might be a good time to update that? Though if updating to v4 is going to require significant changes we might as well wait for #16 like you said

Reply:
I'm planning on working through all of the active approved issues while I'm here, so it'll get looked at soon. But I do think it's out of scope of this PR

Reply:
Sounds good :+1: